### PR TITLE
Initial work

### PR DESCRIPTION
**Pull request by pruzanov**
This workflow is a slightly modified one-step workflow which Yogi wrote, it looks at a couple of fastq files (we need paired-end sequencing data) and checks if
* number of R1 and R2 reads matches
* order is the same
* does optional sorting

### CODE REVIEW

File: checkFastq.wdl [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  parameter_meta {
+    pFastqR1: "R1"
+    pFastqR2: "R2"
+    sample: "Sample prefix"
+    timeout: "Timeout in hours for this task"
+    jobMemory: "Java memory for Picard"
+    modules: "Names and versions of modules needed for variant calling"
+  }
+
+  command <<<
```
Comment (apmasell) [inline checkFastq.wdl:65]: This needs to be moved into a separate script. It is too large to be inline.

File: checkFastq.wdl [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+       if (countR1 != countR2):
+           die('Read count mismatch!')
+
+       """ if headers are different, sort R1 and R2 """
+       if (proc2 != proc):
+           sortfile(r1)
+           sortfile(r2)
+       else:
+           cmd = 'cp ' + r1 + ' ' + sample + '_R1.fastq.gz'
+           os.system(cmd)
```
Comment (apmasell) [inline checkFastq.wdl:99]: These can use `shutil.copy` which is safer.

File: checkFastq.wdl [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+
+   def checkR1R2(r1, r2, sample):
+       """ extract the headers and check if they are the same for R1 and R2 """
+       cmd = 'zcat ' + r1 + ' | paste - - - - | cut -f 1 -d " " ' + '> headerR1'
+       os.system(cmd)
+       cmd = 'zcat ' + r2 + ' | paste - - - - | cut -f 1 -d " " ' + '> headerR2'
+       os.system(cmd)
+
+       proc = subprocess.check_output('md5sum {0}'.format('headerR1'), shell=True).decode('utf-8').rstrip().split()[0]
```
Comment (apmasell) [inline checkFastq.wdl:84]: `shell=True` is insecure. These should be replaced with lists of individual arguments. There's details in the subprocess command.

File: checkFastq.wdl [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+   import sys 
+   def readfile(fastq):
+       fastqfile = gzip.open(fastq, "rb")
+       contents = fastqfile.read()
+       return contents
+
+
+   def checkR1R2(r1, r2, sample):
+       """ extract the headers and check if they are the same for R1 and R2 """
+       cmd = 'zcat ' + r1 + ' | paste - - - - | cut -f 1 -d " " ' + '> headerR1'
```
Comment (apmasell) [inline checkFastq.wdl:79]: The `os.system` call is insecure. This should be redesigned. Moreover, if this is only used to generate an md5sum, a different implementation would be better.

File: checkFastq.wdl [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+           os.system(cmd)
+           cmd = 'cp ' + r2 + ' ' + sample + '_R2.fastq.gz'
+           os.system(cmd)
+
+
+   def sortfile(fastq):
+       out = os.path.basename(fastq)
+       cmd = 'zcat ' + fastq + ' | paste - - - - | sort -k1,1 |tr ' + '"\\t"' + ' "\\n"' + ' | gzip >' + out
+       print(cmd)
+       os.system(cmd)
```
Comment (apmasell) [inline checkFastq.wdl:108]: Again, insecure.

File: checkFastq.wdl [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+       proc2 = subprocess.check_output('md5sum {0}'.format('headerR2'), shell=True).decode('utf-8').rstrip().split()[0]
+
+       countR1 = subprocess.check_output('cat headerR1 | wc -l', shell=True).decode('utf-8').rstrip().split()[0]
+       countR2 = subprocess.check_output('cat headerR2 | wc -l', shell=True).decode('utf-8').rstrip().split()[0]
+
+       if (countR1 != countR2):
+           die('Read count mismatch!')
+
+       """ if headers are different, sort R1 and R2 """
+       if (proc2 != proc):
```
Comment (apmasell) [inline checkFastq.wdl:94]: I'm not sure what the logic is here. This doesn't check that the headers match after sorting.

File: checkFastq.wdl [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    modules: "Names and versions of modules needed for variant calling"
+  }
+
+  command <<<
+   python3<<CODE
+   import gzip
+   import os
+   import subprocess
+   import sys 
+   def readfile(fastq):
```
Comment (apmasell) [inline checkFastq.wdl:71]: Is this function used?

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (apmasell) [inline pom.xml:1]: Since you have the pom file, you might as well run the Vidarr migration script.

File: checkFastq.wdl (hunk 1 above)
Comment (pruzanov) [reply checkFastq.wdl:65]: I actually thought it's a good idea to have it here, it's a very small script. I guess we have different opinions about large and small  :)

File: checkFastq.wdl (hunk 2 above)
Comment (pruzanov) [reply checkFastq.wdl:99]: sure

File: checkFastq.wdl (hunk 4 above)
Comment (pruzanov) [reply checkFastq.wdl:79]: I can always blame Yogi since this is mostly his work untouched

**Pull request merged**